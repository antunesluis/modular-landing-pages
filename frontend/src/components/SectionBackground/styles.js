import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
  background-image: url(${theme.backgroundImage});
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, $background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${$background && containerBackgroundActivate(theme)};

    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
