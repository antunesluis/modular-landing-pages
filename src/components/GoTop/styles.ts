import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};

    border-radius: 0.4rem;
    transition: all 0.2s ease-in-out;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);

    &:hover {
      transform: translateY(-3px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
      background: ${theme.colors.secondaryColor};
    }

    &:active {
      transform: translateY(0);
      box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
      background: ${theme.colors.primaryColor};
    }

    > svg {
      width: 2.4rem;
      height: 2.4rem;
      transition: transform 0.2s ease-in-out;
    }

    &:hover > svg {
      transform: translateY(-3px);
    }

    &:active > svg {
      transform: translateY(0);
    }
  `}
`;
