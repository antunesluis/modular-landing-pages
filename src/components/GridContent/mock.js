export default {
  title: 'Our Digital Solutions',
  html: `
    <h2>Transform Your Business with Our Solutions</h2>
    <p>
      We are experts in developing <strong>innovative</strong> and <strong>customized</strong> 
      solutions for companies looking to stand out in the digital market. Our team combines technical 
      expertise with strategic vision to deliver exceptional results.
    </p>

    <h3>Why choose our company?</h3>

    <ul>
      <li>
        <strong>Proven Experience</strong> - Over 10 years transforming ideas into 
        successful digital solutions
      </li>
      <li>
        <strong>Cutting-edge Technology</strong> - We use the latest market technologies 
        to ensure performance and security
      </li>
      <li>
        <strong>Dedicated Support</strong> - Team always available to assist 
        in all stages of the project
      </li>
    </ul>

    <blockquote>
      "The partnership with the company was fundamental to our business's digital growth. 
      The team is extremely professional and dedicated."
      <cite>- Maria Silva, CEO of TechCorp</cite>
    </blockquote>

    <h3>Our Services</h3>

    <p>
      We offer a complete range of digital services to boost your business:
    </p>

    <table>
      <thead>
        <tr>
          <th>Service</th>
          <th>Benefits</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Web Development</td>
          <td>Responsive, fast, and SEO-optimized websites</td>
        </tr>
        <tr>
          <td>Digital Marketing</td>
          <td>Customized strategies to increase your online presence</td>
        </tr>
        <tr>
          <td>E-commerce</td>
          <td>Robust and secure platforms for online sales</td>
        </tr>
      </tbody>
    </table>

    <h3>Technologies we use</h3>
    <p>
      We work with the most modern technologies in the market, including 
      <code>React</code>, <code>Node.js</code>, and <code>Python</code>. 
      Our stack is constantly updated to ensure the best solutions.
    </p>

    <h3>Contact us</h3>
    <p>
      Want to know more about how we can help your business? 
      <a href="#contact">Click here</a> to speak with one of our consultants or 
      send an email to <a href="mailto:contact@company.com">contact@company.com</a>
    </p>
  `,
};
