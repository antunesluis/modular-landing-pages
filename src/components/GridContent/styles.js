import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.xlarge} 0 ${theme.spacings.xlarge};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    width: 100%;
  `}
`;

export const HtmlContent = styled.div`
  ${({ theme }) => css`
    width: 100%;
    overflow-x: hidden; // Prevent horizontal scroll at component level

    p {
      font-size: ${theme.fonts.sizes.medium};
      line-height: 1.5;
      margin: ${theme.spacings.medium} 0;
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: ${theme.spacings.xlarge} 0 ${theme.spacings.xlarge};
      font-weight: 700;
      font-family: ${theme.fonts.family.secondary};
    }

    // Estilização para links
    a {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    // Estilização para listas
    ul,
    ol {
      margin: ${theme.spacings.small} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
    }

    ul {
      list-style: disc;
    }

    ol {
      list-style: decimal;
    }

    li {
      text-align: left;
    }

    // Estilização para imagens
    img {
      max-width: 100%;
      height: auto;
    }

    // Estilização para citações
    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondaryColor};
      color: ${theme.colors.primaryColor};
      margin: ${theme.spacings.medium} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
      font-style: italic;
    }

    // Estilização para tabelas
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
      display: block;
      overflow-x: auto; // Adiciona scroll horizontal apenas na tabela
      -webkit-overflow-scrolling: touch; // Melhora a rolagem em dispositivos touch

      // Remove display: block em telas maiores
      @media (min-width: 768px) {
        display: table;
      }
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 1px solid ${theme.colors.mediumGray};
      text-align: left;
      min-width: 15rem; // Garante uma largura mínima para células
    }

    table th {
      background: ${theme.colors.primaryColor};
      color: ${theme.colors.white};
    }
  `}
`;
