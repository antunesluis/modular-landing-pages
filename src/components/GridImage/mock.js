export default {
  background: false,
  title: 'Project Gallery',
  description:
    'Discover some of the transformative projects we have developed for our clients. Each solution is unique and designed to maximize results.',
  grid: [
    {
      altText: 'E-commerce Platform',
      large: 'https://picsum.photos/360/360?r=1',
      medium: 'https://picsum.photos/360/360?r=1',
      small: 'https://picsum.photos/360/360?r=1',
      thumbnail: 'https://picsum.photos/360/360?r=1',
    },
    {
      altText: 'Banking App Interface',
      srcImg: 'https://picsum.photos/360/360?r=2',
      medium: 'https://picsum.photos/360/360?r=2',
      small: 'https://picsum.photos/360/360?r=2',
      thumbnail: 'https://picsum.photos/360/360?r=2',
    },
    {
      altText: 'Healthcare Dashboard',
      srcImg: 'https://picsum.photos/360/360?r=3',
      medium: 'https://picsum.photos/360/360?r=3',
      small: 'https://picsum.photos/360/360?r=3',
      thumbnail: 'https://picsum.photos/360/360?r=3',
    },
    {
      altText: 'Education Platform',
      srcImg: 'https://picsum.photos/360/360?r=4',
      medium: 'https://picsum.photos/360/360?r=4',
      small: 'https://picsum.photos/360/360?r=4',
      thumbnail: 'https://picsum.photos/360/360?r=4',
    },
    {
      altText: 'Real Estate Solution',
      srcImg: 'https://picsum.photos/360/360?r=5',
      medium: 'https://picsum.photos/360/360?r=5',
      small: 'https://picsum.photos/360/360?r=5',
      thumbnail: 'https://picsum.photos/360/360?r=5',
    },
    {
      altText: 'Logistics System',
      srcImg: 'https://picsum.photos/360/360?r=6',
      medium: 'https://picsum.photos/360/360?r=6',
      small: 'https://picsum.photos/360/360?r=6',
      thumbnail: 'https://picsum.photos/360/360?r=6',
    },
  ],
};
