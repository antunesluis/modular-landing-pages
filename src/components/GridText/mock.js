export default {
  background: true, // Enables background with theme-defined image
  title: 'Our Services',
  description:
    'We offer complete software development solutions, focusing on quality and innovation. Our specialized team is ready to turn your ideas into reality.',
  grid: [
    {
      title: 'Technical Consulting',
      description:
        'We provide specialized consulting in software architecture, technology choices, and development best practices. We help your team make the best technical decisions.',
    },
    {
      title: 'Web Development',
      description:
        'We create modern and responsive web applications using the latest market technologies. Our focus is on React, Next.js, and Node.js, ensuring performance and scalability for your project.',
    },
    {
      title: 'UX/UI Design',
      description:
        'Our design team creates intuitive and attractive interfaces, focusing on user experience. We work with prototyping, usability testing, and complete design systems.',
    },
  ],
};
