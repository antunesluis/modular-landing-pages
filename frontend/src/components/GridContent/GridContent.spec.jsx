import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { GridContent } from '.';
import mock from './mock';

describe('<GridContent />', () => {
  it('should render grid content', () => {
    const { container } = renderTheme(<GridContent {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render grid content with background', () => {
    const { container } = renderTheme(
      <GridContent {...mock} background={true} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the title', () => {
    renderTheme(<GridContent {...mock} />);
    expect(
      screen.getByRole('heading', { name: mock.title }),
    ).toBeInTheDocument();
  });

  it('should render content', () => {
    renderTheme(<GridContent {...mock} />);
    expect(screen.getByText(/transform your business/i)).toBeInTheDocument();
  });

  it('should match snapshot without background', () => {
    const { container } = renderTheme(
      <GridContent {...mock} background={false} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with background', () => {
    const { container } = renderTheme(
      <GridContent {...mock} background={true} />,
    );
    expect(container).toMatchSnapshot();
  });
});
