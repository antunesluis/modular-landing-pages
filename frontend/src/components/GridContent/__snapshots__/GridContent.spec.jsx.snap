// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridContent /> > should match snapshot with background 1`] = `
.c3 {
  color: #FFFFFF;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background-image: url(/assets/images/backgrounds/blue-bg.svg);
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  color: #FFFFFF;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  padding: 4.0rem 0 4.0rem;
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
  width: 100%;
}

.c4 {
  width: 100%;
  overflow-x: hidden;
}

.c4 p {
  font-size: 2.4rem;
  line-height: 1.5;
  margin: 2.4rem 0;
}

.c4 h1,
.c4 h2,
.c4 h3,
.c4 h4,
.c4 h5,
.c4 h6 {
  margin: 4.0rem 0 4.0rem;
  font-weight: 700;
  font-family: var(--font-montserrat);
}

.c4 a {
  color: #dc143c;
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c4 a:hover {
  filter: brightness(50%);
}

.c4 ul,
.c4 ol {
  margin: 1.6rem 0;
  padding: 2.4rem 3.2rem;
}

.c4 ul {
  list-style: disc;
}

.c4 ol {
  list-style: decimal;
}

.c4 li {
  text-align: left;
}

.c4 img {
  max-width: 100%;
  height: auto;
}

.c4 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #0A1128;
  margin: 2.4rem 0;
  padding: 2.4rem 3.2rem;
  font-style: italic;
}

.c4 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
  display: block;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.c4 table td,
.c4 table th {
  padding: 1.6rem;
  border: 1px solid #DDDDDD;
  text-align: left;
  min-width: 15rem;
}

.c4 table th {
  background: #0A1128;
  color: #FFFFFF;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (min-width: 768px) {
  .c4 table {
    display: table;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Our Digital Solutions
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            
    
            <h2>
              Transform Your Business with Our Solutions
            </h2>
            
    
            <p>
              
      We are experts in developing 
              <strong>
                innovative
              </strong>
               and 
              <strong>
                customized
              </strong>
               
      solutions for companies looking to stand out in the digital market. Our team combines technical 
      expertise with strategic vision to deliver exceptional results.
    
            </p>
            

    
            <h3>
              Why choose our company?
            </h3>
            

    
            <ul>
              
      
              <li>
                
        
                <strong>
                  Proven Experience
                </strong>
                 - Over 10 years transforming ideas into 
        successful digital solutions
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Cutting-edge Technology
                </strong>
                 - We use the latest market technologies 
        to ensure performance and security
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Dedicated Support
                </strong>
                 - Team always available to assist 
        in all stages of the project
      
              </li>
              
    
            </ul>
            

    
            <blockquote>
              
      "The partnership with the company was fundamental to our business's digital growth. 
      The team is extremely professional and dedicated."
      
              <cite>
                - Maria Silva, CEO of TechCorp
              </cite>
              
    
            </blockquote>
            

    
            <h3>
              Our Services
            </h3>
            

    
            <p>
              
      We offer a complete range of digital services to boost your business:
    
            </p>
            

    
            <table>
              
      
              <thead>
                
        
                <tr>
                  
          
                  <th>
                    Service
                  </th>
                  
          
                  <th>
                    Benefits
                  </th>
                  
        
                </tr>
                
      
              </thead>
              
      
              <tbody>
                
        
                <tr>
                  
          
                  <td>
                    Web Development
                  </td>
                  
          
                  <td>
                    Responsive, fast, and SEO-optimized websites
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    Digital Marketing
                  </td>
                  
          
                  <td>
                    Customized strategies to increase your online presence
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    E-commerce
                  </td>
                  
          
                  <td>
                    Robust and secure platforms for online sales
                  </td>
                  
        
                </tr>
                
      
              </tbody>
              
    
            </table>
            

    
            <h3>
              Technologies we use
            </h3>
            
    
            <p>
              
      We work with the most modern technologies in the market, including 
      
              <code>
                React
              </code>
              , 
              <code>
                Node.js
              </code>
              , and 
              <code>
                Python
              </code>
              . 
      Our stack is constantly updated to ensure the best solutions.
    
            </p>
            

    
            <h3>
              Contact us
            </h3>
            
    
            <p>
              
      Want to know more about how we can help your business? 
      
              <a
                href="#contact"
              >
                Click here
              </a>
               to speak with one of our consultants or 
      send an email to 
              <a
                href="mailto:contact@company.com"
              >
                contact@company.com
              </a>
              
    
            </p>
            
  
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContent /> > should match snapshot without background 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  padding: 4.0rem 0 4.0rem;
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
  width: 100%;
}

.c4 {
  width: 100%;
  overflow-x: hidden;
}

.c4 p {
  font-size: 2.4rem;
  line-height: 1.5;
  margin: 2.4rem 0;
}

.c4 h1,
.c4 h2,
.c4 h3,
.c4 h4,
.c4 h5,
.c4 h6 {
  margin: 4.0rem 0 4.0rem;
  font-weight: 700;
  font-family: var(--font-montserrat);
}

.c4 a {
  color: #dc143c;
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c4 a:hover {
  filter: brightness(50%);
}

.c4 ul,
.c4 ol {
  margin: 1.6rem 0;
  padding: 2.4rem 3.2rem;
}

.c4 ul {
  list-style: disc;
}

.c4 ol {
  list-style: decimal;
}

.c4 li {
  text-align: left;
}

.c4 img {
  max-width: 100%;
  height: auto;
}

.c4 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #0A1128;
  margin: 2.4rem 0;
  padding: 2.4rem 3.2rem;
  font-style: italic;
}

.c4 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
  display: block;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.c4 table td,
.c4 table th {
  padding: 1.6rem;
  border: 1px solid #DDDDDD;
  text-align: left;
  min-width: 15rem;
}

.c4 table th {
  background: #0A1128;
  color: #FFFFFF;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (min-width: 768px) {
  .c4 table {
    display: table;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Our Digital Solutions
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            
    
            <h2>
              Transform Your Business with Our Solutions
            </h2>
            
    
            <p>
              
      We are experts in developing 
              <strong>
                innovative
              </strong>
               and 
              <strong>
                customized
              </strong>
               
      solutions for companies looking to stand out in the digital market. Our team combines technical 
      expertise with strategic vision to deliver exceptional results.
    
            </p>
            

    
            <h3>
              Why choose our company?
            </h3>
            

    
            <ul>
              
      
              <li>
                
        
                <strong>
                  Proven Experience
                </strong>
                 - Over 10 years transforming ideas into 
        successful digital solutions
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Cutting-edge Technology
                </strong>
                 - We use the latest market technologies 
        to ensure performance and security
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Dedicated Support
                </strong>
                 - Team always available to assist 
        in all stages of the project
      
              </li>
              
    
            </ul>
            

    
            <blockquote>
              
      "The partnership with the company was fundamental to our business's digital growth. 
      The team is extremely professional and dedicated."
      
              <cite>
                - Maria Silva, CEO of TechCorp
              </cite>
              
    
            </blockquote>
            

    
            <h3>
              Our Services
            </h3>
            

    
            <p>
              
      We offer a complete range of digital services to boost your business:
    
            </p>
            

    
            <table>
              
      
              <thead>
                
        
                <tr>
                  
          
                  <th>
                    Service
                  </th>
                  
          
                  <th>
                    Benefits
                  </th>
                  
        
                </tr>
                
      
              </thead>
              
      
              <tbody>
                
        
                <tr>
                  
          
                  <td>
                    Web Development
                  </td>
                  
          
                  <td>
                    Responsive, fast, and SEO-optimized websites
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    Digital Marketing
                  </td>
                  
          
                  <td>
                    Customized strategies to increase your online presence
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    E-commerce
                  </td>
                  
          
                  <td>
                    Robust and secure platforms for online sales
                  </td>
                  
        
                </tr>
                
      
              </tbody>
              
    
            </table>
            

    
            <h3>
              Technologies we use
            </h3>
            
    
            <p>
              
      We work with the most modern technologies in the market, including 
      
              <code>
                React
              </code>
              , 
              <code>
                Node.js
              </code>
              , and 
              <code>
                Python
              </code>
              . 
      Our stack is constantly updated to ensure the best solutions.
    
            </p>
            

    
            <h3>
              Contact us
            </h3>
            
    
            <p>
              
      Want to know more about how we can help your business? 
      
              <a
                href="#contact"
              >
                Click here
              </a>
               to speak with one of our consultants or 
      send an email to 
              <a
                href="mailto:contact@company.com"
              >
                contact@company.com
              </a>
              
    
            </p>
            
  
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContent /> > should render grid content 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  padding: 4.0rem 0 4.0rem;
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
  width: 100%;
}

.c4 {
  width: 100%;
  overflow-x: hidden;
}

.c4 p {
  font-size: 2.4rem;
  line-height: 1.5;
  margin: 2.4rem 0;
}

.c4 h1,
.c4 h2,
.c4 h3,
.c4 h4,
.c4 h5,
.c4 h6 {
  margin: 4.0rem 0 4.0rem;
  font-weight: 700;
  font-family: var(--font-montserrat);
}

.c4 a {
  color: #dc143c;
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c4 a:hover {
  filter: brightness(50%);
}

.c4 ul,
.c4 ol {
  margin: 1.6rem 0;
  padding: 2.4rem 3.2rem;
}

.c4 ul {
  list-style: disc;
}

.c4 ol {
  list-style: decimal;
}

.c4 li {
  text-align: left;
}

.c4 img {
  max-width: 100%;
  height: auto;
}

.c4 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #0A1128;
  margin: 2.4rem 0;
  padding: 2.4rem 3.2rem;
  font-style: italic;
}

.c4 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
  display: block;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.c4 table td,
.c4 table th {
  padding: 1.6rem;
  border: 1px solid #DDDDDD;
  text-align: left;
  min-width: 15rem;
}

.c4 table th {
  background: #0A1128;
  color: #FFFFFF;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (min-width: 768px) {
  .c4 table {
    display: table;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Our Digital Solutions
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            
    
            <h2>
              Transform Your Business with Our Solutions
            </h2>
            
    
            <p>
              
      We are experts in developing 
              <strong>
                innovative
              </strong>
               and 
              <strong>
                customized
              </strong>
               
      solutions for companies looking to stand out in the digital market. Our team combines technical 
      expertise with strategic vision to deliver exceptional results.
    
            </p>
            

    
            <h3>
              Why choose our company?
            </h3>
            

    
            <ul>
              
      
              <li>
                
        
                <strong>
                  Proven Experience
                </strong>
                 - Over 10 years transforming ideas into 
        successful digital solutions
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Cutting-edge Technology
                </strong>
                 - We use the latest market technologies 
        to ensure performance and security
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Dedicated Support
                </strong>
                 - Team always available to assist 
        in all stages of the project
      
              </li>
              
    
            </ul>
            

    
            <blockquote>
              
      "The partnership with the company was fundamental to our business's digital growth. 
      The team is extremely professional and dedicated."
      
              <cite>
                - Maria Silva, CEO of TechCorp
              </cite>
              
    
            </blockquote>
            

    
            <h3>
              Our Services
            </h3>
            

    
            <p>
              
      We offer a complete range of digital services to boost your business:
    
            </p>
            

    
            <table>
              
      
              <thead>
                
        
                <tr>
                  
          
                  <th>
                    Service
                  </th>
                  
          
                  <th>
                    Benefits
                  </th>
                  
        
                </tr>
                
      
              </thead>
              
      
              <tbody>
                
        
                <tr>
                  
          
                  <td>
                    Web Development
                  </td>
                  
          
                  <td>
                    Responsive, fast, and SEO-optimized websites
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    Digital Marketing
                  </td>
                  
          
                  <td>
                    Customized strategies to increase your online presence
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    E-commerce
                  </td>
                  
          
                  <td>
                    Robust and secure platforms for online sales
                  </td>
                  
        
                </tr>
                
      
              </tbody>
              
    
            </table>
            

    
            <h3>
              Technologies we use
            </h3>
            
    
            <p>
              
      We work with the most modern technologies in the market, including 
      
              <code>
                React
              </code>
              , 
              <code>
                Node.js
              </code>
              , and 
              <code>
                Python
              </code>
              . 
      Our stack is constantly updated to ensure the best solutions.
    
            </p>
            

    
            <h3>
              Contact us
            </h3>
            
    
            <p>
              
      Want to know more about how we can help your business? 
      
              <a
                href="#contact"
              >
                Click here
              </a>
               to speak with one of our consultants or 
      send an email to 
              <a
                href="mailto:contact@company.com"
              >
                contact@company.com
              </a>
              
    
            </p>
            
  
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContent /> > should render grid content with background 1`] = `
.c3 {
  color: #FFFFFF;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background-image: url(/assets/images/backgrounds/blue-bg.svg);
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  color: #FFFFFF;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  padding: 4.0rem 0 4.0rem;
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
  width: 100%;
}

.c4 {
  width: 100%;
  overflow-x: hidden;
}

.c4 p {
  font-size: 2.4rem;
  line-height: 1.5;
  margin: 2.4rem 0;
}

.c4 h1,
.c4 h2,
.c4 h3,
.c4 h4,
.c4 h5,
.c4 h6 {
  margin: 4.0rem 0 4.0rem;
  font-weight: 700;
  font-family: var(--font-montserrat);
}

.c4 a {
  color: #dc143c;
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c4 a:hover {
  filter: brightness(50%);
}

.c4 ul,
.c4 ol {
  margin: 1.6rem 0;
  padding: 2.4rem 3.2rem;
}

.c4 ul {
  list-style: disc;
}

.c4 ol {
  list-style: decimal;
}

.c4 li {
  text-align: left;
}

.c4 img {
  max-width: 100%;
  height: auto;
}

.c4 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #0A1128;
  margin: 2.4rem 0;
  padding: 2.4rem 3.2rem;
  font-style: italic;
}

.c4 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
  display: block;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

.c4 table td,
.c4 table th {
  padding: 1.6rem;
  border: 1px solid #DDDDDD;
  text-align: left;
  min-width: 15rem;
}

.c4 table th {
  background: #0A1128;
  color: #FFFFFF;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (min-width: 768px) {
  .c4 table {
    display: table;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Our Digital Solutions
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            
    
            <h2>
              Transform Your Business with Our Solutions
            </h2>
            
    
            <p>
              
      We are experts in developing 
              <strong>
                innovative
              </strong>
               and 
              <strong>
                customized
              </strong>
               
      solutions for companies looking to stand out in the digital market. Our team combines technical 
      expertise with strategic vision to deliver exceptional results.
    
            </p>
            

    
            <h3>
              Why choose our company?
            </h3>
            

    
            <ul>
              
      
              <li>
                
        
                <strong>
                  Proven Experience
                </strong>
                 - Over 10 years transforming ideas into 
        successful digital solutions
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Cutting-edge Technology
                </strong>
                 - We use the latest market technologies 
        to ensure performance and security
      
              </li>
              
      
              <li>
                
        
                <strong>
                  Dedicated Support
                </strong>
                 - Team always available to assist 
        in all stages of the project
      
              </li>
              
    
            </ul>
            

    
            <blockquote>
              
      "The partnership with the company was fundamental to our business's digital growth. 
      The team is extremely professional and dedicated."
      
              <cite>
                - Maria Silva, CEO of TechCorp
              </cite>
              
    
            </blockquote>
            

    
            <h3>
              Our Services
            </h3>
            

    
            <p>
              
      We offer a complete range of digital services to boost your business:
    
            </p>
            

    
            <table>
              
      
              <thead>
                
        
                <tr>
                  
          
                  <th>
                    Service
                  </th>
                  
          
                  <th>
                    Benefits
                  </th>
                  
        
                </tr>
                
      
              </thead>
              
      
              <tbody>
                
        
                <tr>
                  
          
                  <td>
                    Web Development
                  </td>
                  
          
                  <td>
                    Responsive, fast, and SEO-optimized websites
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    Digital Marketing
                  </td>
                  
          
                  <td>
                    Customized strategies to increase your online presence
                  </td>
                  
        
                </tr>
                
        
                <tr>
                  
          
                  <td>
                    E-commerce
                  </td>
                  
          
                  <td>
                    Robust and secure platforms for online sales
                  </td>
                  
        
                </tr>
                
      
              </tbody>
              
    
            </table>
            

    
            <h3>
              Technologies we use
            </h3>
            
    
            <p>
              
      We work with the most modern technologies in the market, including 
      
              <code>
                React
              </code>
              , 
              <code>
                Node.js
              </code>
              , and 
              <code>
                Python
              </code>
              . 
      Our stack is constantly updated to ensure the best solutions.
    
            </p>
            

    
            <h3>
              Contact us
            </h3>
            
    
            <p>
              
      Want to know more about how we can help your business? 
      
              <a
                href="#contact"
              >
                Click here
              </a>
               to speak with one of our consultants or 
      send an email to 
              <a
                href="mailto:contact@company.com"
              >
                contact@company.com
              </a>
              
    
            </p>
            
  
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
